crates/subspace-networking/src/node_runner.rs

```diff
@@ -161,17 +160,14 @@ impl NodeRunner {
         if !self.reserved_peers.is_empty() {
             trace!(%local_peer_id, "Checking reserved peers connection: {:?}", self.reserved_peers);
 
-            let reserved_peers = convert_multiaddresses(self.reserved_peers.clone())
-                .into_iter()
-                .collect::<HashMap<_, _>>();
             let connected_peers_id_set = connected_peers.iter().cloned().collect();
-            let reserved_peers_id_set = reserved_peers.keys().cloned().collect::<HashSet<_>>();
+            let reserved_peers_id_set = self.reserved_peers.keys().cloned().collect::<HashSet<_>>();
 
             let missing_reserved_peer_ids =
                 reserved_peers_id_set.difference(&connected_peers_id_set);
 
             for peer_id in missing_reserved_peer_ids {
-                if let Some(addr) = reserved_peers.get(peer_id) {
+                if let Some(addr) = self.reserved_peers.get(peer_id) {
```

Comment:
Hmm, I believe you can `unwrap` here, as difference only returns ids stored in `reserved_peers`

Reply:
`unwrap` relies on the current state of code that can change any moment. I'd prefer to keep static guarantees of no panic.